### PR TITLE
IMathics in Mathics repository

### PR DESCRIPTION
**Pull request by poke1024**
Changes `setup.py` to install the IMathics package and jupyter kernel if the option `--jupyter` is given, i.e. `python setup.py install --jupyter`. Seems to work fine.

Still needs work on the Jupyter test cases.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -143,6 +147,80 @@ def run(self):
...
+            kernel_spec_manager.install_kernel_spec(
+                kernel_spec_path,
+                kernel_name=kernel_json['name'],
+                user=not self.user)
+        except Exception as e:
+            log.error(str(e.args))
+            log.error('Failed to install kernel spec')
+
+
+CMDCLASS['install'] = install
```
Comment (poke1024) [inline setup.py]: OK, great.

**Comment by JoseCarlosGarcia95** [comment]
I have checked IMathics and IMathics does not work with the latest version of Mathics, and hasn't got any update since two years.
I'm working on a new jupyter kernel that works with the latest mathics release, and I'm working actively adding new functions to it.

https://github.com/JoseCarlosGarcia95/mathicsjupyternotebook

File: setup.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (GarkGarcia) [reply setup.py:219]: @poke1024 Has this been addressed already?

**Comment by GarkGarcia** [comment]
> I have checked IMathics and IMathics does not work with the latest version of Mathics, and hasn't got any update since two years.

We have been facing some governance issues 😅️ Essentially Mathics has been dormant since 2018. Would you be interested in merging your recent (if we can call 2018 "recent") work into this repository?

**Comment by slel** [comment]
There's also @mmatera's IWolfram:
https://github.com/mmatera/iwolfram

**Comment by GarkGarcia** [comment]
@slel @mmatera I'd like your input on this issue, I'm not sure what's the best solution in here. On the one hand this PR passes all tests, so I see no reason to reject it. On the other hand @mmatera's iwolfram is probably more up to date (and there could be incompatibilities between this PR in the work in mmatera/iwolfram).

**Comment by mmatera** [comment]
I think that IMathics broke the django's interface - after you merged it it stop working with the error:

"""
Open the graphical user interface at
http://localhost:8000
in Firefox, Chrome, or Safari to use Mathics

loading symbols
Internal Server Error: /
Traceback (most recent call last):
  File "/opt/miniconda3/site-packages/Django-1.11.29-py3.6.egg/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/opt/miniconda3/site-packages/Django-1.11.29-py3.6.egg/django/core/handlers/base.py", line 249, in _legacy_get_response
    response = self._get_response(request)
  File "/opt/miniconda3/site-packages/Django-1.11.29-py3.6.egg/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/opt/miniconda3/site-packages/Django-1.11.29-py3.6.egg/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/opt/miniconda3/site-packages/Mathics-1.1.dev0-py3.6.egg/mathics/web/views.py", line 67, in main_view
    return render(request, 'main.html', context)
  File "/opt/miniconda3/site-packages/Django-1.11.29-py3.6.egg/django/shortcuts.py", line 30, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "/opt/miniconda3/site-packages/Django-1.11.29-py3.6.egg/django/template/loader.py", line 67, in render_to_string
    template = get_template(template_name, using=using)
  File "/opt/miniconda3/site-packages/Django-1.11.29-py3.6.egg/django/template/loader.py", line 25, in get_template
    raise TemplateDoesNotExist(template_name, chain=chain)
django.template.exceptions.TemplateDoesNotExist: main.html
[09/Sep/2020 03:36:02] "GET / HTTP/1.1" 500 88689
Not Found: /favicon.ico
[09/Sep/2020 03:36:02] "GET /favicon.ico HTTP/1.1" 404 4140
"""


On the other hand, iwolfram is an independent project, for a jupyter kernel which interacts between jupyter web interface and the command line mathics kernel (or the wolfram kernel) in text mode. I guess that the best would be to suggest in the mathics documentation to install iwolfram as an alternative of the current, standard django's interface.

**Comment by mmatera** [comment]
I would keep the functionality of Manipulate also as a pymathics package, and move its implementation to the IMathics project. Then, we could maintain both subprojects separately.

**Comment by GarkGarcia** [comment]
> I would keep the functionality of Manipulate also as a pymathics package, and move its implementation to the IMathics project. Then, we could maintain both subprojects separately.

I'm inclined to go with option too.

**Comment by mmatera** [comment]
I close this PR since we didn't get any new feedback, and goes against the trend of the modularization of the project. The glue between IMathics and Mathics (for example, to implement Manipulate[]) is going now to be implemented through the Pymathics module mechanism.

**Pull request closed without merging**